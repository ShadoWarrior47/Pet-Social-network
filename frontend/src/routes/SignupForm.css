/* General container styling */
.Signup-container div {
    max-width: 600px !important;
    margin: 0px auto;
    padding: 10px;
    font-family: Arial, sans-serif;
    min-width: 500px;
}

.Signup-container form {
    display: flex;
    flex-direction: column;
    gap: 10px;
    padding: 10px;
    background-color: #f9f9f9;
    border: 1px solid #ddd;
    border-radius: 5px;
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
}

.Signup-container h1 {
    text-align: center;
    margin-bottom: 20px;
    font-size: 24px;
    color: #333;
}

.Signup-container input {
    width: 100%;
    padding: 5px;
    font-size: 16px;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-shadow: inset 0 1px 3px rgba(0, 0, 0, 0.1);
    outline: none;
    transition: border-color 0.3s;
}

.Signup-container input:focus {
    border-color: #007bff;
}

.Signup-container label {
    font-size: 14px;
    font-weight: bold;
    margin-bottom: 2px;
    color: #555;
    text-align: center;
}

.Signup-container button {
    padding: 10px 20px;
    font-size: 16px;
    color: white;
    background-color: #007bff;
    border: none;
    border-radius: 4px;
    cursor: pointer;
    transition: background-color 0.3s, transform 0.2s;
    width: 150px;
    margin: 0px auto;
}

.Signup-container button:hover {
    background-color: #0056b3;
}

.Signup-container button:active {
    transform: scale(0.98);
}