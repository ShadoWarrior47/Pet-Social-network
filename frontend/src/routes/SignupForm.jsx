import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import "./SignupForm.css";

function SignupForm({ signup }) {
  const navigate = useNavigate();
  const INITIAL_STATE = {
    username: '',
    password: '',
    name: '',
    email: '',
    profilePic: '',
    isAdmin: false
  }

  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value
    }));
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    await signup(formData);
    navigate("/");
  };

  return (
    <div className="Signup-container">
      <h1>Sign Up</h1>
      <form onSubmit={handleSubmit}>

        <div>
          <label htmlFor="username">
            Username :
          </label>
          <input
            id="username"
            name="username"
            type="text"
            placeholder="Username"
            autoComplete="username"
            value={formData.username}
            onChange={handleChange}
            required
          />
        </div>

        <div>
          <label htmlFor="password">
            Password :
          </label>
          <input
            id="password"
            name="password"
            type="password"
            placeholder="Password"
            autoComplete="new-password"
            value={formData.password}
            onChange={handleChange}
            required
          />
        </div>

        <div>
          <label htmlFor="name">
            Name :
          </label>
          <input
            id="name"
            name="name"
            type="text"
            placeholder="Name"
            value={formData.name}
            onChange={handleChange}
            required
          />
        </div>

        <div>
          <label htmlFor="email">
            E-mail :
          </label>
          <input
            id="email"
            name="email"
            type="text"
            placeholder="email"
            autoComplete="email"
            value={formData.email}
            onChange={handleChange}
            required
          />
        </div>

        <div>
          <label htmlFor="profilePic">
            Profile Picture Url :
          </label>
          <input
            id="profilePic"
            name="profilePic"
            type="text"
            placeholder="Profile Picture Url"
            value={formData.profilePic}
            onChange={handleChange}
          />
        </div>

        <input type="hidden" name="isAdmin" value={formData.isAdmin} />

        <button type="submit">Sign up</button>
      </form>
    </div>

  );
}

export default SignupForm;
