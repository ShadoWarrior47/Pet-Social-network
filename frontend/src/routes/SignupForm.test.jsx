import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import SignupForm from "./SignupForm";
import { vi } from "vitest";

it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <SignupForm signup={() => { }} />
        </MemoryRouter>
    );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <SignupForm signup={() => { }} />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("runs the signup function on form submit", async function () {
    const signupMock = vi.fn();
    const { getByLabelText, getByText } = render(
        <MemoryRouter>
            <SignupForm signup={signupMock} />
        </MemoryRouter>
    );

    const usernameInput = getByLabelText("Username :");
    const passwordInput = getByLabelText("Password :");
    const nameInput = getByLabelText("Name :");
    const emailInput = getByLabelText("E-mail :");
    const profilePicInput = getByLabelText("Profile Picture Url :");
    const submitButton = getByText("Sign up");

    fireEvent.change(usernameInput, { target: { value: "testuser" } });
    fireEvent.change(passwordInput, { target: { value: "password123" } });
    fireEvent.change(nameInput, { target: { value: "Test" } });
    fireEvent.change(emailInput, { target: { value: "testuser@example.com" } });
    fireEvent.change(profilePicInput, { target: { value: "http://example.com/profile.jpg" } });


    fireEvent.click(submitButton);

    expect(signupMock).toHaveBeenCalledWith({
        username: "testuser",
        password: "password123",
        name: "Test",
        email: "testuser@example.com",
        profilePic: "http://example.com/profile.jpg",
        isAdmin: false,
    });
});
