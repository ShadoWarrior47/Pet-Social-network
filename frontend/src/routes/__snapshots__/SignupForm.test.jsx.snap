// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Signup-container"
  >
    <h1>
      Sign Up
    </h1>
    <form>
      <div>
        <label
          for="username"
        >
          Username :
        </label>
        <input
          autocomplete="username"
          id="username"
          name="username"
          placeholder="Username"
          required=""
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="password"
        >
          Password :
        </label>
        <input
          autocomplete="new-password"
          id="password"
          name="password"
          placeholder="Password"
          required=""
          type="password"
          value=""
        />
      </div>
      <div>
        <label
          for="name"
        >
          Name :
        </label>
        <input
          id="name"
          name="name"
          placeholder="Name"
          required=""
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          E-mail :
        </label>
        <input
          autocomplete="email"
          id="email"
          name="email"
          placeholder="email"
          required=""
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="profilePic"
        >
          Profile Picture Url :
        </label>
        <input
          id="profilePic"
          name="profilePic"
          placeholder="Profile Picture Url"
          type="text"
          value=""
        />
      </div>
      <input
        name="isAdmin"
        type="hidden"
        value="false"
      />
      <button
        type="submit"
      >
        Sign up
      </button>
    </form>
  </div>
</DocumentFragment>
`;
